samples/Mvc.FileUpload/Filters/GenerateAntiforgeryTokenCookieForAjax.cs

```diff
@@ -0,0 +1,22 @@
+﻿using Microsoft.AspNetCore.Antiforgery;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Mvc.FileUpload.Filters
+{
+    public class GenerateAntiforgeryTokenCookieForAjaxAttribute : ActionFilterAttribute
+    {
+        public override void OnActionExecuted(ActionExecutedContext context)
+        {
+            var antiforgery = context.HttpContext.RequestServices.GetService<IAntiforgery>();
+
+            // We can send the request token as a JavaScript-readable cookie, and Angular will use it by default.
+            var tokens = antiforgery.GetAndStoreTokens(context.HttpContext);
+            context.HttpContext.Response.Cookies.Append(
+                "XSRF-TOKEN",
+                tokens.RequestToken,
+                new CookieOptions() { HttpOnly = false });
```

Comment:
Question: An application could have a mix of pages where some use Ajax and some regular form post. Since this code sets a cookie which would be sent in subsequent requests to all pages of the app, I was initially wondering if doing a form post to a page would fail antiforgery validation (I say this because we recently made a change to look at the header first and then the form field). I tried testing few scenarios and the form post seems to succeed without any issue even though the value of the token is different between the header and the form field token. I am not sure if I am missing any scenarios. 
OR would it be safe to set the `CookieOptions.Path` to the url that we are posting to so that the scope of this cookie is limited only to that action.